src/main/java/picard/analysis/CollectQualityYieldMetrics.java

```diff
@@ -103,7 +107,11 @@ protected boolean usesNoRefReads() {
     @Override
     protected void setup(final SAMFileHeader header, final File samFile) {
         IOUtil.assertFileIsWritable(OUTPUT);
-        this.collector = new QualityYieldMetricsCollector(USE_ORIGINAL_QUALITIES, INCLUDE_SECONDARY_ALIGNMENTS, INCLUDE_SUPPLEMENTAL_ALIGNMENTS);
+        if (FLOW_MODE) {
+            this.collector = new QualityYieldMetricsCollector(USE_ORIGINAL_QUALITIES, INCLUDE_SECONDARY_ALIGNMENTS, INCLUDE_SUPPLEMENTAL_ALIGNMENTS, true);
```

Comment:
can you replace these 5 lines with 1 by simply passing "FLOW_MODE" to the constructor?